examples/Azure_IoT_Adu_ESP32/Azure_IoT_Adu_ESP32.ino

```diff
@@ -0,0 +1,1024 @@
+  if (az_result_failed(rc))
+  {
+    Logger.Info("Failed to get request id");
+  }
+
+  return az_span_slice(out_span, 0, az_span_size(out_span) - az_span_size(remainder));
+}
+
+static void prvParseAduUrl(az_span xUrl, az_span* pxHost, az_span* pxPath)
+{
+  xUrl = az_span_slice_to_end(xUrl, sizeof("http://") - 1);
+  int32_t lPathPosition = az_span_find(xUrl, AZ_SPAN_FROM_STR("/"));
+  *pxHost = az_span_slice(xUrl, 0, lPathPosition);
+  *pxPath = az_span_slice_to_end(xUrl, lPathPosition);
+}
+
+void update_started() { Serial.println("CALLBACK:  HTTP update process started"); }
+
+void update_finished() { Serial.println("CALLBACK:  HTTP update process finished"); }
+
+void update_progress(int cur, int total)
+{
+  Serial.printf("CALLBACK:  HTTP update process at %d of %d bytes...\n", cur, total);
+}
+
+void update_error(int err) { Serial.printf("CALLBACK:  HTTP update fatal error code %d\n", err); }
+
+static az_result download_and_write_to_flash(void)
+{
+  az_span url_host_span;
+  az_span url_path_span;
+  prvParseAduUrl(adu_update_request.file_urls[0].url, &url_host_span, &url_path_span);
+
+  WiFiClient client;
+
+  httpUpdate.onStart(update_started);
+  httpUpdate.onEnd(update_finished);
+  httpUpdate.onProgress(update_progress);
+  httpUpdate.onError(update_error);
+  httpUpdate.rebootOnUpdate(false);
+
+  // TODO: Add SHA check of image (don't immediately reboot)
+
+  /* TODO: remove this hack. */
+  char null_terminated_host[128];
+  (void)memcpy(null_terminated_host, az_span_ptr(url_host_span), az_span_size(url_host_span));
+  null_terminated_host[az_span_size(url_host_span)] = '\0';
+
+  /* TODO: remove this hack. */
+  char null_terminated_path[128];
+  (void)memcpy(null_terminated_path, az_span_ptr(url_path_span), az_span_size(url_path_span));
+  null_terminated_path[az_span_size(url_path_span)] = '\0';
+
+  t_httpUpdate_return ret
+      = httpUpdate.update(client, null_terminated_host, 80, null_terminated_path);
+
+  if (ret != HTTP_UPDATE_OK)
+  {
+    Logger.Error("Image download failed");
+    return AZ_ERROR_CANCELED;
+  }
+  else
+  {
+    Logger.Info("Download of image succeeded");
+    return AZ_OK;
+  }
+}
+
+static az_result verify_image(az_span sha256_hash, int32_t update_size)
+{
+  az_result result;
+  esp_err_t espErr;
+
+  int32_t out_size;
+  int32_t read_size;
+  az_span out_buffer = AZ_SPAN_FROM_BUFFER(adu_sha_buffer);
+
+  Logger.Info("Verifying downloaded image with manifest SHA256 hash");
+
+  const esp_partition_t* current_partition = esp_ota_get_running_partition();
+  const esp_partition_t* update_partition = esp_ota_get_next_update_partition(current_partition);
+
+  if (az_result_failed(result = az_base64_decode(out_buffer, sha256_hash, &out_size)))
+  {
+    Logger.Error("az_base64_decode failed: core error=0x" + String(result, HEX));
+  }
+  else
+  {
+    Logger.Info("Unencoded the base64 encoding\r\n");
+    result = AZ_OK;
+  }
+
+  if (result != AZ_OK)
```

Comment:
Can the contents of this if statement be moved to within the if statement at line 281?